corehq/apps/app_manager/views/forms.py

```diff
@@ -755,6 +754,10 @@ def is_valid(value):
         }
     else:
         refs = json.loads(data.get('case_references', '{}'))
-    if set(refs) - {"load"} or not all(is_valid(v) for v in refs["load"].values()):
+
+    try:
+        references = CaseReferences.wrap(copy.deepcopy(refs))
```

Comment:
good call, updated and added a test to convince myself. https://github.com/dimagi/commcare-hq/pull/14710/commits/237e1f89e82ce3a79cb4ece4d7da3a6f1e70eee5